feat: install by store paths (#2413)
flox/flox: cli/flox/doc/manifest.toml.md

    If an attrpath is specified, it is checked whether
    `packages.$system.$attrpath` or `legacyPackages.$system.$attrpath` exist.

#### Store paths

> Flake descriptors allow installing software from an arbitrary Nix flake.
> 
> The full list of flake descriptor options is:

Reviewer: ```suggestion
Store path descriptors allow installing software from an arbitrary Nix store path.

The full list of store path descriptor options is:
```

```
Descriptor ::= {
  store-path         = STRING
, systems            = null | [<STRING>, ...]
, priority           = null | <INT>
}
```

Only `store-path` is required.

> `priority` behave the same as described above for catalog

Reviewer: ```suggestion
`priority` behaves the same as described above for catalog
```

descriptors and flake installables, and `store-path` is described below:

> `store-path`
> :   Specifies a nix store path, i.e. a nix built pacakge in `/nix/store`.
>     This can be the result of a native Nix operations,
>     such as `nix build`, `nix copy`, etc.
>     To build an environment that references store paths,
>     the store path has to be available to nix on the current system.
>     Without distributing the store path first via nix tooling,
>     the environment will fail to build.
>     As such this feature is most suitable for local experiments and
>     ad-hoc interoperability with Nix.
> `system`
> :   Behaves equally to the system attribute of catalog descriptors
>     and flakes installables.
>     Unlike the former, users are encouraged to specify it,
>     because store paths are generally system dependent.

Reviewer: Combined suggestions to capitalise Nix, fix a typo, change the flow of the sentences, and ensure that `system` gets rendered correctly.
```suggestion
`store-path`
:   Specifies a nix store path, i.e. a nix built package in `/nix/store`. This
    can be the result of a native Nix operations such as `nix build`, `nix
    copy`, etc.
    The store path has to be available on the current system in order to build
    the environment. The environment will fail to build on other systems without
    first distributing the store path via Nix tooling.
    As such, this feature is most suitable for local experiments and ad-hoc
    interoperability with Nix.

`system`
:   Behaves equally to the system attribute of catalog descriptors
    and flakes installables.
    Unlike the former, users are encouraged to specify it,
    because store paths are generally system dependent.
```
